src/cache-loan.js

```diff
@@ -0,0 +1,27 @@
+const _merge = require('lodash.merge')
+const Loan = require('./loan')
+
+class CacheLoan {
```

Comment:
Trying to figure out if the `CacheLoan` class should extend the `Loan` class or not

Reply:
My thinking is that it can be instantiated at the start of a Lambda as something like `let Loan = new CacheLoan(table, region)`, then it works in much the same way as ActiveRecord, so you can do `Loan.new`, `Loan.create` etc

Reply:
That way it doesn't directly extend the `Loan` class, so you can't create a `Loan` object as `new CacheLoan`, rather that creates an API for working with `Loan` objects.

I think it would be cleaner to have it all bundled into the `Loan` class, but I can't think of how to separately set the tableName and region for the loans to be cached to. This way the potential exists to have several `CacheLoans` which specify different tables or regions in the same code, although this isn't required for current purposes.

Reply:
Thinking about it, it may be better for the `Loan` class to be completely agnostic of what database it uses, and simply provide a `Loan#setDB` (or similar) method to pass it a database instance. The `CacheLoan` class can then instantiate a database class on it's constructor, and pass this to any `Loans` created by it.

This way if ever multiple `Loans` exist concurrently, there is one shared database object between them, instead of them each instantiating their own. Again, this isn't strictly required for current purposes, as I can't think of an example where one Lambda would create more than one `Loan` object.

Reply:
See https://gist.github.com/DomBelcher/e2e081c81b847e182513a05aa82345f5 for an example of my thinking

Reply:
Similarly `Request` and `User` could follow this pattern, although this may then lead to an overabundance of classes for a fairly simple task

Reply:
https://gist.github.com/DomBelcher/e16d749d3f12eca7c32aeb26511640ab is a more concrete version of how I'm thinking of the classes being structured